lib/grpc/client/adapters/gun.ex

```diff
@@ -135,7 +135,37 @@ defmodule GRPC.Client.Adapters.Gun do
   end
 
   @impl true
-  def recv_headers(%{conn_pid: conn_pid}, %{stream_ref: stream_ref}, opts) do
+  def receive_data(
+        %{server_stream: true, channel: %{adapter_payload: adapter_payload}, payload: payload} =
+          stream,
+        opts
+      ) do
```

Comment:
Let's split this match, leaving only the flow-control at the function head
```suggestion
        %{server_stream: true} = stream,
        opts
      ) do
```

And then match to extract the data below. Same for the other clause